akka-typed-tests/src/test/java/akka/typed/javadsl/BehaviorBuilderTest.java

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright (C) 2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.typed.javadsl;
+
+import akka.typed.Behavior;
+import akka.typed.Terminated;
+import org.junit.Test;
+import org.scalatest.junit.JUnitSuite;
+
+import java.util.ArrayList;
+
+import static akka.typed.javadsl.Actor.same;
+import static akka.typed.javadsl.Actor.stopped;
+
+public class BehaviorBuilderTest extends JUnitSuite {
+    interface Message {
+    }
+
+    static class One implements Message {
+        public String foo() {
+          return "Bar";
+        }
+    }
+    static class MyList<T> extends ArrayList<T> implements Message {
+
+    };
+
+    @Test
+    public void shouldCompile() {
+        Behavior<Message> b = BehaviorBuilder.<Message>apply()
+                .message(One.class, o -> o.foo().startsWith("a"), (ctx, o) -> same())
+                .message(One.class, (ctx, o) -> {
+                    o.foo();
+                    return same();
+                })
+                .messageUnchecked(MyList.class, (ActorContext<Message> ctx, MyList<String> l) -> {
+                  String first = l.get(0);
+                  return Actor.<Message>same();
+                })
+                .signal(Terminated.class, (ctx, t) -> {
+                    System.out.println("Terminating along with " + t.ref());
+                    return stopped();
+                });
+    }
+
+    @Test
+    public void shouldBeUsableToCreateFromActor() {
+      Behavior<Message> b = Actor.<Message>immutable()
+              .message(One.class, (ctx, o) -> {
+                o.foo();
+                return same();
+              })
+              .signal(Terminated.class, (ctx, t) -> {
+                System.out.println("Terminating along with " + t.ref());
+                return stopped();
+              });
```

Comment:
pretty nice, I wonder if those would be easier to understand if called onMessage / onSignal hm... I guess others will have opinions on this too